### ISSUE TITLE
Serving pretrained models in production

### ISSUE THREAD

**Issue by yoldemir** [UKPLab/sentence-transformers#267]
Does anyone have an example of how to serve the pretrained models (such as bert-base-nli-stsb-mean-tokens) in production? I'd like to ideally serve this from Java. If that's not natively possible, using curl might be an option too.

**Comment by bmwas** [comment]
The newest version of elasticsearch has a "dense vector" field that can allow you to save documents together with their Sentence BERT embeddings and serve the model in production. 
For more - visit 
- https://www.elastic.co/guide/en/elasticsearch/reference/current/dense-vector.html
- https://www.elastic.co/blog/text-similarity-search-with-vectors-in-elasticsearch

**Comment by yoldemir** [comment]
Thank you I'll check this out, even though I was hoping to do this directly using PyTorch without an additional dependency.

**Comment by yoldemir** [comment]
I'm trying to see if I can export a pretrained model as a TorchScript model (in .pt1 format) and then use https://github.com/pytorch/java-demo but I haven't had success in creating a TorchScript model using tracing or annotation. If I can get it to work I'll post an example here for others. In the meantime, I'd appreciate if anyone has any pointers about exporting these models as TorchScript models.

**Comment by nreimers** [comment]
Hi @yoldemir 
The easiest solution would be to create a REST api, that accepts the sentence and returns the vector.

You could use flask / werkzeug / FastAPI  to create such REST api. With FastAPI, it should only be a few lines of code to get a REST API up and running.

**Comment by yoldemir** [comment]
Certainly. TorchScript seems to load and execute models without a Python runtime available, and I was hoping to get that behavior. However there are some limitations currently blocking that, like assuming no control flow in the forward() method when you use tracing, and assuming no keyword argument expansion when you use annotation. I'll keep tinkering to see if I can get it to work, otherwise I might have to fall back to a REST API like you suggested.

**Comment by kaismh** [comment]
I am also interested in inference from Java, https://github.com/deepjavalibrary/djl looks like a good option. However, it only supports TorchScript at this time, https://docs.djl.ai/docs/pytorch/how_to_convert_your_model_to_torchscript.html

**Comment by soheilesm** [comment]
@kaismh @yoldemir did you guys have any success with converting SBert into an executable via torchscript?

I tried to use Bert itself - I could do it for the embedding generation part, but not 'yet' for the tokenizer part - Currently looking into it, maybe we can share with each other if we make it happen?

**Comment by dabasmoti** [comment]
@soheilesm Did you success with converting SBert into an executable via torchscript?
